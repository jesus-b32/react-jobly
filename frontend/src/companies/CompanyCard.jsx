import React from 'react';
import { Card, CardBody, CardTitle, CardText } from 'reactstrap';

function CompanyCard({ company }) {
  return (
    <>
      <Card className='mb-4'>
        <CardBody className='text-center'>
          <CardTitle tag={'h4'}>{company.name}</CardTitle>
          <CardText>{company.description}</CardText>
        </CardBody>
      </Card>
    </>
  );
}

export default CompanyCard;
