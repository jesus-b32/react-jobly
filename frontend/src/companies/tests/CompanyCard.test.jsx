import CompanyCard from '../CompanyCard';
import { MemoryRouter } from 'react-router-dom';
import { companyData } from '../../tests/testCommon';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';

it('should render without crashing', () => {
  const company = companyData[0];
  render(
    <MemoryRouter>
      <CompanyCard company={company} />
    </MemoryRouter>
  );
});

it('should match snapshot', () => {
  const company = companyData[0];
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard company={company} />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});
