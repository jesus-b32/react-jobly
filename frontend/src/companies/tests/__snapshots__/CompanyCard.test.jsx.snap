// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-4 card"
  >
    <div
      class="text-center card-body"
    >
      <h4
        class="card-title"
      >
        test company 1
      </h4>
      <p
        class="card-text"
      >
        this is a test
      </p>
    </div>
  </div>
</DocumentFragment>
`;
